pkg/insights/insightsuploader/insightsuploader.go

```diff
@@ -157,6 +159,7 @@ func (c *Controller) Run(ctx context.Context) {
 			}
 			c.reporter.SetSafeInitialStart(false)
 			klog.V(4).Infof("Uploaded report successfully in %s", time.Now().Sub(start))
+			c.archiveUploaded <- struct{}{}
```

Comment:
Would would happen when the gaherer was not consuming from this channel and should the uploader care?

Reply:
As far as I can tell about Golang channels, there is no problem if nobody reads the channel, but maybe I'm wrong

Reply:
The problem is it would get stuck on trying to write

Reply:
https://gobyexample.com/non-blocking-channel-operations